docs/data/date-pickers/custom-field/custom-field.md

```diff
@@ -154,3 +154,309 @@ and you don't want the UI to look like a Text Field, you can replace the field w
+  const [inputValue, setInputValue] = useInputValue(value, format);
+
+  const placeholder = useParsedFormat(props);
+
+  const { hasValidationError, getValidationErrorForNewValue } = useValidation({
+    value,
+    timezone,
+    props,
+    validator: validateDate,
+  });
+
+  const handleChange = (event) => {
+    const newInputValue = event.target.value;
+    const newValue = dayjs(newInputValue, format);
+    setInputValue(newInputValue);
+    onChange(newValue, { validationError: getValidationErrorForNewValue(newValue) });
+  };
+
+  return (
+    <TextField
+      placeholder={format}
+      value={inputValue}
+      onChange={handleChange}
+      error={hasValidationError}
+    />
+  );
+}
+```
+
+#### Spread props to the DOM
+
+The field receives a lot of props that cannot be forwarded to the DOM element without warnings.
+You can use the `useSplitFieldProps` hook to get the props that can be forwarded safely to the DOM:
+
+```tsx
+const { internalProps, forwardedProps } = useSplitFieldProps(
+  // The props received by the field component
+  props,
+  // The value type ("date", "time" or "date-time")
+  'date',
+);
+
+return (
+  <TextField {...forwardedProps} value={inputValue} onChange={handleChange}>
+)
+```
+
+:::success
+The `forwardedProps` contain props like `slots`, `slotProps` and `sx` that are specific to MUI.
+You can omit them if the component your are forwarding the props to does not support those concepts:
+
+```jsx
+const { slots, slotProps, sx, ...other } = props;
+
+const { internalProps, forwardedProps } = useSplitFieldProps(other, 'date');
+
+return (
+  <TextField {...forwardedProps} value={inputValue} onChange={handleChange}>
+)
+```
+
+:::
+
+##### Updated example
+
+Here is the updated example with the props forwarded to the DOM:
+
+```jsx
+function CustomDateField(props) {
+  // TextField does not support slots and slotProps before `@mui/material` v6.0
+  const { slots, slotProps, ...other } = props;
+  const { internalProps, forwardedProps } = useSplitFieldProps(other, 'date');
+
+  const { format, timezone, value, onChange } = internalProps;
+  const [inputValue, setInputValue] = useInputValue(value, format);
+
+  const { hasValidationError, getValidationErrorForNewValue } = useValidation({
+    value,
+    timezone,
+    props: internalProps,
+    validator: validateDate,
+  });
+
+  const handleChange = (event) => {
+    const newInputValue = event.target.value;
+    const newValue = dayjs(newInputValue, format);
+    setInputValue(newInputValue);
+    onChange(newValue, { validationError: getValidationErrorForNewValue(newValue) });
+  };
+
+  const placeholder = useParsedFormat(internalProps);
+
+  return (
+    <TextField
+      {...forwardedProps}
+      placeholder={placeholder}
+      value={inputValue}
+      onChange={handleChange}
+      error={hasValidationError}
+    />
+  );
+}
+```
+
+#### Enhanced basic example
+
+Here is a live demo of the example created in all the previous sections:
```

Comment:
WFYT about avoiding duplication here? 🤔 
I feel like having the final live example here should be enough. 🤞 

Besides, both section headings sound a bit misaligned with the presented information. 🙈 
Maybe we could stick with just one heading, i.e.: Full live example or Full custom example.